src/devices/Pn532/Pn532.cs

```diff
@@ -0,0 +1,1623 @@
+            //var ret = SetSAM();
+            LogInfo.Log($"Setting SAM changed: {ret}", LogLevel.Info);
+            // Check the version
+            if (!CheckVersion())
+                throw new Exception("Can't find a PN532");
+            // Apply default parameters
+            ret = SetParameters(ParametersFlags.fAutomaticATR_RES | ParametersFlags.fAutomaticRATS);
+            LogInfo.Log($"Setting Parameters Falgs changed: {ret}", LogLevel.Info);
+        }
+
+        /// <summary>
+        /// Create a PN532 using SPI
+        /// </summary>
+        /// <param name="spiDevice"></param>
+        public Pn532(SpiDevice spiDevice)
+        {
+#if DEBUG
+            LogLevel = LogLevel.Debug;
+#endif
+
+            _spiDevice = spiDevice;
+            _controller = new GpioController();
+            _controller.OpenPin(_pin, PinMode.Output);
+            _controller.Write(_pin, PinValue.High);
+            Thread.Sleep(2);
+            LogLevel = LogLevel.Debug;
+            WakeUp();
+            var ret = SetSecurityAccessModule();
+            //var ret = SetSAM();
+            LogInfo.Log($"Setting SAM changed: {ret}", LogLevel.Info);
+            // Check the version
+            if (!CheckVersion())
+                throw new Exception("Can't find a PN532");
+            // Apply default parameters
+            ret = SetParameters(ParametersFlags.fAutomaticATR_RES | ParametersFlags.fAutomaticRATS);
+            LogInfo.Log($"Setting Parameters Falgs changed: {ret}", LogLevel.Info);
+            ret = SetSecurityAccessModule();
+            //var ret = SetSAM();
+            LogInfo.Log($"Setting SAM changed: {ret}", LogLevel.Info);
+        }
+
+        /// <summary>
+        /// Create a PN532 using I2C
+        /// </summary>
+        /// <param name="i2CDevice"></param>
+        public Pn532(I2cDevice i2CDevice)
+        {            
+#if DEBUG
+            LogLevel = LogLevel.Debug;
+#endif
+
+            _i2cDevice = i2CDevice;
+            WakeUp();
+            var ret = SetSecurityAccessModule();
+            //var ret = SetSAM();
+            LogInfo.Log($"Setting SAM changed: {ret}", LogLevel.Info);
+            // Check the version
+            if (!CheckVersion())
+                throw new Exception("Can't find a PN532");
+            // Apply default parameters
+            ret = SetParameters(ParametersFlags.fAutomaticATR_RES | ParametersFlags.fAutomaticRATS);
+            LogInfo.Log($"Setting Parameters Falgs changed: {ret}", LogLevel.Info);
+            ret = SetSecurityAccessModule();
+            //var ret = SetSAM();
+            LogInfo.Log($"Setting SAM changed: {ret}", LogLevel.Info);
+        }
+
+        /// <summary>
+        /// Run self tests
+        /// Note: some self tests are not implemented yet
+        /// </summary>
+        /// <param name="diagnoseMode">The self test to run</param>
+        /// <returns>True when success</returns>
+        public bool RunSelfTest(DiagnoseMode diagnoseMode)
```

Comment:
I think this should return SelfTestStatus or throw an exception with useful info. I.e. what should user do when this returns false?

Reply:
This function will have to evolve once advance tests will be implemented. Rational not to throw exception is to be able to run the tests in a raw and see the results.
Some tests will require to return a more complex object like the PollingTestToTarget one. I will leave it so far as not really the priority for the moment.